src/chronify/store.py

```diff
@@ -32,48 +53,145 @@ def __init__(
         else:
             self._engine = engine
 
+        self._schema_mgr = SchemaManager(self._engine, self._metadata)
+        if self._engine.url.database != ":memory:":
+            self.update_sqlalchemy_metadata()
+
+    @classmethod
+    def create_in_memory_db(
+        cls,
+        engine_name: str = "duckdb",
+        **connect_kwargs: Any,
+    ) -> "Store":
+        """Create a Store with an in-memory database."""
+        return Store(engine=create_engine(f"{engine_name}:///:memory:", **connect_kwargs))
+
+    @classmethod
+    def create_file_db(
+        cls,
+        file_path: Path | str = "time_series.db",
+        engine_name: str = "duckdb",
+        overwrite: bool = False,
+        **connect_kwargs: Any,
+    ) -> "Store":
+        """Create a Store with a file-based database."""
+        path = file_path if isinstance(file_path, Path) else Path(file_path)
+        if path.exists():
+            if overwrite:
+                path.unlink()
+            else:
+                msg = f"{path} already exists. Choose a different path or set overwrite=True."
+                raise InvalidOperation(msg)
+        return Store(engine=create_engine(f"{engine_name}:///{path}", **connect_kwargs))
+
+    @classmethod
+    def load_from_file(
+        cls,
+        file_path: Path | str,
+        engine_name: str = "duckdb",
+        **connect_kwargs: Any,
+    ) -> "Store":
+        """Load an existing store from a database."""
+        path = file_path if isinstance(file_path, Path) else Path(file_path)
+        if not path.exists():
+            msg = str(path)
+            raise FileNotFoundError(msg)
+        return Store(engine=create_engine(f"{engine_name}:///{path}", **connect_kwargs))
+
+    def get_table(self, name: str) -> Table:
+        """Return the sqlalchemy Table object."""
+        if not self.has_table(name):
+            msg = f"{name=}"
+            raise TableNotStored(msg)
+        return Table(name, self._metadata)
+
+    def has_table(self, name: str) -> bool:
+        """Return True if the database has a table with the given name."""
+        return name in self._metadata.tables
+
+    def list_tables(self) -> list[str]:
+        """Return a list of user tables in the database."""
+        return [x for x in self._metadata.tables if x != SchemaManager.SCHEMAS_TABLE]
+
+    def try_get_table(self, name: str) -> Table | None:
+        """Return the sqlalchemy Table object or None if it is not stored."""
+        if not self.has_table(name):
+            return None
+        return Table(name, self._metadata)
+
+    def update_sqlalchemy_metadata(self) -> None:
+        """Update the sqlalchemy metadata for table schema. Call this method if you add tables
+        in the sqlalchemy engine outside of this class.
+        """
+        self._metadata.reflect(self._engine, views=True)
+
+    def backup(self, dst: Path | str, overwrite: bool = False) -> None:
```

Comment:
This is a temporary solution that is not well-thought out. I submitted #28 to fix it later.